apps/server/src/modules/board/saga/copy-room-boards.step.ts

```diff
@@ -58,26 +59,53 @@ export class CopyRoomBoardsStep extends SagaStep<'copyRoomBoards'> {
 		const sourceRoom = await this.roomService.getSingleRoom(sourceRoomId);
 		const targetRoom = await this.roomService.getSingleRoom(targetRoomId);
 
-		const sourceStorageLocationReference = { id: sourceRoom.schoolId, type: StorageLocation.SCHOOL };
-		const targetStorageLocationReference = { id: targetRoom.schoolId, type: StorageLocation.SCHOOL };
+		const copyStatuses: CopyStatus[] = [];
+		for (const board of boards) {
+			const copyStatus = await this.columnBoardService.copyColumnBoard({
+				originalColumnBoardId: board.id,
+				targetExternalReference: {
+					type: BoardExternalReferenceType.Room,
+					id: targetRoomId,
+				},
+				sourceStorageLocationReference: { id: sourceRoom.schoolId, type: StorageLocation.SCHOOL },
+				targetStorageLocationReference: { id: targetRoom.schoolId, type: StorageLocation.SCHOOL },
+				userId,
+				targetSchoolId: targetRoom.schoolId,
+				copyTitle: board.title,
+			});
+			copyStatuses.push(copyStatus);
+		}
 
-		const copyStatuses = await Promise.all(
-			boards.map((board) =>
-				this.columnBoardService.copyColumnBoard({
-					originalColumnBoardId: board.id,
-					targetExternalReference: {
-						type: BoardExternalReferenceType.Room,
-						id: targetRoomId,
-					},
-					sourceStorageLocationReference,
-					targetStorageLocationReference,
-					userId,
-					targetSchoolId: targetRoom.schoolId,
-					copyTitle: board.title,
-				})
-			)
-		);
+		const status: CopyStatus = {
+			title: 'board',
+			type: CopyElementType.ROOM,
+			status: this.copyHelperService.deriveStatusFromElements(copyStatuses),
+			copyEntity: targetRoom,
+			originalEntity: sourceRoom,
+			elements: copyStatuses,
+		};
+
+		await this.swapLinkedIdsInBoards(status);
 
 		return copyStatuses;
 	}
+
+	// TODO similar method already in learnroom module, consider refactoring for keeping DRY
+	private async swapLinkedIdsInBoards(copyStatus: CopyStatus): Promise<CopyStatus> {
+		const map = new Map<EntityId, EntityId>();
+		const copyDict = this.copyHelperService.buildCopyEntityDict(copyStatus);
+		copyDict.forEach((value, key) => map.set(key, value.id));
+
+		const elements = copyStatus.elements ?? [];
+		const updatedElements: CopyStatus[] = [];
+		for (const el of elements) {
```

Comment:
Why not use `Promise.all` here like in the similar method in the learnroom module?